specs/WebAuthenticationRequested.md

```diff
@@ -0,0 +1,280 @@
+```c#
+webView.CoreWebView2.WebAuthenticationRequested += delegate (object sender, CoreWebView2WebAuthenticationRequestedEventArgs args)
+{
+    if (args.Challenge.Equals("Expected login credentials")) {
+        args.Cancel = true;
+    }
+};
+```
+
+# Remarks
+
+
+# API Notes
+
+
+# API Details
+```idl
+/// WebView2 enables you to host web content using the latest Microsoft Edge
+/// browser and web technology.
+
+[uuid(76eceacb-0462-4d94-ac83-423a6793775e), object, pointer_default(unique)]
+interface ICoreWebView2 : IUnknown
+{
+  /// ...
+
+  /// Add an event handler for the WebAuthenticationRequested event.
+  /// WebAuthenticationRequested event is raised when WebView encountered a Basic HTTP
+  /// Authentication request as described in
+  /// https://developer.mozilla.org/en-US/docs/Web/HTTP/Authentication.
+  ///
+  /// The host can provide a response with credentials for the authentication or
+  /// cancel the request. If the host doesn't set the Cancel property to true or
+  /// set the Response property, then WebView2 will show the default
+  /// authorization challenge dialog prompt to the user.
+  ///
+  HRESULT add_WebAuthenticationRequested(
+    [in] ICoreWebView2WebAuthenticationRequestedEventHandler* eventHandler,
+    [out] EventRegistrationToken* token);
+  /// Remove an event handler previously added with add_WebResourceRequested.
+  HRESULT remove_WebAuthenticationRequested(
+      [in] EventRegistrationToken token);
+}
+
+/// This is the CoreWebView2WebAuthenticationRequestedEventHandler interface
+[uuid(f87e5d35-3248-406b-81dd-1c36aab8081d), object, pointer_default(unique)]
+interface ICoreWebView2WebAuthenticationRequestedEventHandler : IUnknown
+{
+  /// Called to provide the implementer with the event args for the
+  /// corresponding event.
+  HRESULT Invoke(
+      [in] ICoreWebView2* sender,
+      [in] ICoreWebView2WebAuthenticationRequestedEventArgs* args);
+}
+
+/// Represents a Basic HTTP authentication response that contains a user name
+/// and a password as according to RFC7617 (https://tools.ietf.org/html/rfc7617)
+[uuid(bc9cfd60-29c4-4943-a83b-d0d2f3e7df03), object, pointer_default(unique)]
+interface ICoreWebView2WebAuthenticationResponse : IUnknown
+{
+  /// User name provided for authorization.
+  [propget] HRESULT UserName([out, retval] LPWSTR* userName);
+  /// Set user name property
+  [propput] HRESULT UserName([in] LPCWSTR userName);
+
+  /// Password provided for authorization
+  [propget] HRESULT Password([out, retval] LPWSTR* password);
+  /// Set password property
+  [propput] HRESULT Password([in] LPCWSTR password);
+}
+
+/// Event args for the WebAuthorizationRequested event. Will contain the
+/// request that led to the HTTP authorization challenge, the challenge
+/// and allows the host to provide authentication response or cancel the request.
+[uuid(51d3adaa-159f-4e48-ad39-a86beb2c1435), object, pointer_default(unique)]
+interface ICoreWebView2WebAuthenticationRequestedEventArgs : IUnknown
```

Comment:
This should have a GetDeferral right? I imagine a common case would be to prompt the user and that would require async UI